offlineimap: add proxy dependency

Vonfry opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

Nobody

Commit:
offlineimap: add proxy dependency

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review-may-2019/3032/97

nlewo commented:
Since the proxy is a standard configuration option, I think this dependency should be added by default. Could you compare the closure size before and after adding it to the derivation (`nix path-info -S ./result`)?

nlewo commented on pkgs/tools/networking/offlineimap/default.nix:
```diff
@@ -23,7 +23,10 @@ python2Packages.buildPythonApplication rec {
   nativeBuildInputs = [ asciidoc libxml2 libxslt docbook_xsl ];
-  propagatedBuildInputs = with python2Packages; [ six kerberos rfc6555 ];
+  propagatedBuildInputs = with python2Packages; [ six kerberos rfc6555 ]
+                                             ++ (if enableProxy
```
FYI, it would be better to use `lib.optional` instead of `if` (but I think pysocks should be unconditionally added).

nlewo reviewed (changes requested):
cf my previous comment also.

Commit:
fix: offlineimap add pysocks, take adive from
https://github.com/NixOS/nixpkgs/pull/76347#issuecomment-569022837, add
pysocks without option.

Vonfry commented:
A print by `nix path-info -S` before and after: 
```
/nix/store/z891lj9c7fdgznnrgqnsmnjdmp4q6nmy-offlineimap-7.3.0	  102404368
/nix/store/h1d7q8r65h2k8wd6vvhixv65q6yk49hw-offlineimap-7.3.0	  102517808
```

Ma27 commented:
> A print by nix path-info -S before and after:

While keeping an eye on the closure-size is a good thing to avoid a bloated system-closure, I don't think that an increase of 3MB for an end-user application are a bad thing.

Vonfry commented:
Thanks, I have added pysocks to the buildInputs instead of an option.

nlewo commented:
Yep, 3MB is definitely not a issue.
Thx!
